### PR TITLE
Re-enable DeprecationCreationPerformanceTest and use new API

### PR DESCRIPTION
**Pull request by blindpirate**
The old DeprecationCreationPerformanceTest is too flaky, now let's
try latest deprecation API.

### CODE REVIEW

**Comment by blindpirate** [comment]
@big-guy When do you think we can remove the old API?

**Comment by blindpirate** [comment]
@bot-gradle test this

**Comment by blindpirate** [comment]
After switching to new deprecation API, it seems better.

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/DeprecationCreationPerformanceTest.groovy [hunk 1]
```diff
@@ -17,9 +17,7 @@
 package org.gradle.performance.regression.corefeature
 
 import org.gradle.performance.AbstractCrossVersionPerformanceTest
-import spock.lang.Ignore
 
-@Ignore('https://github.com/gradle/gradle-private/issues/3202')
 class DeprecationCreationPerformanceTest extends AbstractCrossVersionPerformanceTest {
```
Comment (wolfs) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/DeprecationCreationPerformanceTest.groovy:21]: You need to increase the minimum base version to a version including the new deprecation API.

File: subprojects/performance/src/templates/generateLotsOfDeprecationWarnings/build.gradle [hunk 2]
```diff
@@ -17,13 +16,21 @@
 
 @groovy.transform.CompileStatic
 void createDeprecations(int iterations) {
-    for (int i=0; i<iterations; i++) {
+    for (int i = 0; i < iterations; i++) {
         // every 10th deprecation should have a different message
-        if(i%10 == 0) {
-            org.gradle.util.DeprecationLogger.nagUserOfDeprecated("Some unique deprecation No #\$i")
```
Comment (wolfs) [inline subprojects/performance/src/templates/generateLotsOfDeprecationWarnings/build.gradle:23]: Can you as well delete `org.gradle.util.DeprecationLogger` from our code base?

**Review by wolfs — CHANGES_REQUESTED** [review]

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/DeprecationCreationPerformanceTest.groovy (hunk 1 above)
Comment (blindpirate) [reply subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/DeprecationCreationPerformanceTest.groovy:21]: Ah, thanks for reminder.

File: subprojects/performance/src/templates/generateLotsOfDeprecationWarnings/build.gradle (hunk 2 above)
Comment (blindpirate) [reply subprojects/performance/src/templates/generateLotsOfDeprecationWarnings/build.gradle:23]: Shall we remove it immediately or wait for 7.0? I'm not sure how community members use this API, but myself uses it in my opensource plugin...

Comment (wolfs) [reply subprojects/performance/src/templates/generateLotsOfDeprecationWarnings/build.gradle:23]: Hmmm...

It is an internal API, so we could change it however we want. Do we have any failing smoke tests when we remove the old deprecation API? Do we need to add a deprecation warning when someone uses the old API?

**Pull request merged**